### PR TITLE
SOL-487 Add SearchInitializer

### PR DESCRIPTION
**Pull request by dino-cikatic**
@martynjames this is a base class to support SOL-487 on edx-platform.

### CODE REVIEW

**Comment by martynjames** [comment]
:+1: This looks lovely - just what Dr. Andy ordered! @mattdrayer - any chance you can be 2nd reviewer herein?

File: search/initializer.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+class SearchInitializer(object):
+
+    """
+    Class to set starting environment parameters for search app.
+    Users of this search app will override this class and update setting for SEARCH_INITIALIZER
+    """
+
+    # disabling pylint violations because overriders will want to use these
+    # pylint: disable=unused-argument, no-self-use
```
Comment (mattdrayer) [inline search/initializer.py]: Are these skips supposed to be enabled?  if not I would just remove them.

File: search/tests/test_course_discovery.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 # Some of the subclasses that get used as settings-overrides will yield this pylint
 # error, but they do get used when included as part of the override_settings
 # pylint: disable=too-few-public-methods
+# pylint: disable=too-many-ancestors
```
Comment (mattdrayer) [inline search/tests/test_course_discovery.py]: Commented-out pylints again -- remove if not necessary

**Comment by mattdrayer** [comment]
LGTM!  :+1:

**Pull request merged**